portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -71,15 +81,42 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String message = request.getParameter("message");
         String date = messageDate.toString();
 
+        //Prevents users from posting empty comments by checking the length of the username and message Strings:
         if(username.length() > 1 && message.length() > 1){
 
+            //Analise sentiment of the comment:
+
+            //Create a text document containing the message text:
+            Document commentDoc = Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+
+            //Intantiate a Language Service Client:
+            LanguageServiceClient languageService = LanguageServiceClient.create();
+
+            //Create a sentiment object with the analyzed text document:
+            Sentiment sentiment = languageService.analyzeSentiment(commentDoc).getDocumentSentiment();
+
+            //Get the sentiment score (Negative -1.0 to 1.0 Positive):
+            float score = sentiment.getScore();
+
+            //Close connection to the Language Service Client:
+            languageService.close();
+
+            //Get emoji based on score:
+
+            String emoji = emojify(score);
+
+            //Get a datastore instance:
             DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
+            //Create a Comment Entity with the values of the request:
             Entity commentEntity = new Entity("Comment");
             commentEntity.setProperty("username", username);
             commentEntity.setProperty("message", message);
             commentEntity.setProperty("date", date);
+            commentEntity.setProperty("score", score);
+            commentEntity.setProperty("emoji", emoji);
 
+            //PUT the Entity in the datastore:
```

Comment:
Maybe `PUT` (all caps) is a bit confusing here, I think just "Put" makes more sense? Since it's a method on the datastore API, and not a `PUT` HTTP request, which is usually all caps.

Reply:
Okey! I changed it.